File: Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i=0, max = 0, count = 1;
        while(i<s.length()){
            int count = 0;
            unordered_map<char,int> myMap;
            for(int j=i; j<s.length(); j++){
                if(myMap[s[j]] > 0){
                    if(count> max)
                        max = count;
                    break;
                }
                else{
                    myMap[s[j]]++;
                    count++;
                }
            }
           if(count> max)
                max = count; 
            i++;
        }
        return max;
    }
};

File: Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        unique = defaultdict(int)
        left = 0
        maxlen = 0

        for right in range(len(s)):
            unique[s[right]] += 1
            while unique[s[right]] > 1:
                unique[s[left]] -= 1
                left += 1
            maxlen = max(maxlen, right-left+1)
        
        return maxlen
